src/grain/harvest.js

```diff
@@ -0,0 +1,220 @@
+// @flow
+
+/**
+ * In SourceCred, projects regularly distribute Grain to contributors based on
+ * their Cred scores. This is called a "Harvest".
+ *
+ * This module contains the logic for calculating harvest amounts for
+ * contributors. A harvest contains "receipts" showing how much Grain each
+ * contributor will receive, the timestamp of the harvest in question, and the
+ * "strategy" used to distribute grain.
+ *
+ * Currently we support two strategies:
+ * - FAST, which distributes a fixed amount of grain based on the cred scores
+ * in the most recent completed time interval
+ * - FAIR, which distributes a fixed amount of grain based on cred scores
+ * across
```

Comment:
Second naming issue: fair is too subjective.

Worst case scenario, someone could feel unfairly treated because of this strategy. The name would add insult to injury. Especially if this term shows up in your grain account every week.

Even fast / slow has this issue to some degree as well. If the speed of distribution causes you financial troubles, being confronted with "fast / slow" could be frustrating too. Something objective (like "weekly / lifetime") would be much safer.

Reply:
Yea I agree that fair isn't the best wording, makes it seem like the other strategies might be "unfair".

My issue with "weekly" is that different communities might decide to distribute their harvest at different intervals (monthly, biweekly, daily, etc). What its really doing is paying out "only the last interval" and nothing else.

"Fast" also doesn't convey that IMO. what about "lastIntervalOnly" or "sinceLastHarvest"?

Instead of "fair", could do "mostUnderpaidFirst" ?

Reply:
Yeah I agree weekly/lifetime aren't great names either. Just used them as example to contrast subjective with objective.

I did some musing on the objective form here: https://github.com/sourcecred/sourcecred/pull/1709#pullrequestreview-381093238

> There's also the strategy names. Most ironically `underpayment`.
> To adopt an underpayment strategy means to intentionally underpay, the exact opposite :laughing:.
> 
> There's also important details missing, like units: `fixedAmount`, amount of what?
> And time span: two of three strategies consider all-time cred, which ones?
> 
> If you'd like to move away from giving the strategy nicknames (like fast / slow / velocity based) to a description of what it's doing, you wind up with:
> - `shareByRelativeCredLastInterval`
> - `ensureMinimumLifetimeGrainDistributedToCredRatio`
> - `shareByLifetimeRelativeCredCompensatingUnderpaymentFirst` where underpayment is defined in comment, and share implies we'll use up the whole budget even if the "fairness" was already on target.
> 
> These are pretty obnoxious, so we can push some detail to the comments.
> - `shareByLastIntervalCred`, assume it's relative and linear.
> - `ensureLifetimeGrainToCredRatio` imply minimum by ensure, commenting we don't burn any grain for too high ratios. Also leave out the distinction between transferred and distributed grain.
> - `compensateLifetimeUnderpaymentFirst`, by "first" implying there's an entire second part to this we're leaving out, of sharing the rest of the budget by relative lifetime cred, even if there's no underpayment to compensate.
> 
> These are still large, but at least descriptive. Giving them nicknames again might not be a bad idea.
> 
> Or to split them into less complex, composable functions. Such as the `compensateLifetimeUnderpaymentFirst` could be a combination of:
> 1. `compensateLifetimeUnderpayment`, not "first" because we'll only distribute till everyone is on the fair level, not use up the whole budget even if it's already on target.
> 2. `shareByLifetimeCred`, using the remainder of the budget, if any, after step 1.

Reply:
Hmm, maybe 'Lifetime' and 'Immediate' Distributions, then? 

I like the idea of having short catch names, at least for the user facing concept.

Reply:
"Lifetime" doesn't convey what it's really doing, its more like "most underpaid" since someone could have earned a lot of XP this week but never got paid before so they are the most underpaid. The "lifetime" aspect is only a side effect / part of the calculation of figuring out who is the most underpaid. If the word "underpaid" has a negative connotation, we can try using something else like "Balanced", which also hints at the fact that its trying to optimize for people who's Cred to Grain ratio is "imbalanced", and might be a more neutral term than "Fair".

"Immediate" sounds pretty good to me

Reply:
"Balanced" works for me too.